x-pack/legacy/plugins/transform/public/app/common/aggregations.ts

```diff
@@ -4,9 +4,16 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { composeValidators, patternValidator } from '../../../../ml/common/util/validators';
+
 export type AggName = string;
 
 export function isAggName(arg: any): arg is AggName {
   // allow all characters except `[]>` and must not start or end with a space.
-  return /^[^\s^\[\]>][^\[\]>]+[^\s^\[\]>]$/.test(arg);
+  const validatorFn = composeValidators(
+    patternValidator(/^[^\s]/),
+    patternValidator(/[^\s]$/),
```

Comment:
total nit, but this two could be combined `/^[^\s]+$/`

Reply:
actually it couldn't, `avg responsetime` won't pass validation for instance, but it should